compositor_render/src/renderer.rs

```diff
@@ -57,12 +60,20 @@ impl Renderer {
 
     /// This is very much a work in progress.
     /// For now it just takes a random frame from the input and returns it
-    pub fn render(&self, inputs: HashMap<u32, Frame>) -> Result<Frame, RendererRenderError> {
-        inputs
-            .values()
-            .next()
-            .cloned()
-            .ok_or(RendererRenderError::NoInput(0)) // 0 as a placeholder for now until this is implemented
+    pub fn render(&self, inputs: InputFrames) -> Result<Frame, RendererRenderError> {
+        let output_frame = inputs.frames.values().next().cloned();
+
+        match output_frame {
+            Some(frame) => {
+                let frame = Frame {
+                    data: frame.data.clone(),
+                    resolution: frame.resolution,
+                    pts: inputs.pts,
+                };
+                Ok(frame)
+            }
+            None => Err(RendererRenderError::NoInput(0)),
+        }
```

Comment:
Doesn't this do exactly the same thing?

Reply:
But now with two clones instead of one?

Reply:
> Doesn't this do exactly the same thing?

Now the frame has assigned pts of `InputFrames` (the batch). Without it, we would send to the encoder a frame with pts determined by the encoder. If the queue will send the same frame twice, this will pass it to the encoder, which will. crash on `non monotonically increasing timestamps`.

I haven't optimized this part, since it will be completely changed anyway.

Reply:
oh, I see. makes sense.